source/val/validate_cfg.cpp

```diff
@@ -155,6 +155,27 @@ spv_result_t ValidateBranchConditional(ValidationState_t& _,
   return SPV_SUCCESS;
 }
 
+spv_result_t ValidateSwitch(ValidationState_t& _, const Instruction* inst) {
+  const auto num_operands = inst->operands().size();
+  // At least two operands (selector, default), any more than that are
+  // literal/target.
+
+  // target operands must be OpLabel
+  for (auto i = 2; i < num_operands; i += 2) {
+    const auto literal_ = inst->GetOperandAs<uint32_t>(i + 0);
+    const auto id = inst->GetOperandAs<uint32_t>(i + 1);
+
+    const auto target = _.FindDef(id);
+    if (!target || SpvOpLabel != target->opcode()) {
```

Comment:
Hey @alan-baker are there other places in the validator where branches are checked.  All branches need to target another block in the same function.  For example.

Reply:
This look like the right spot. OpBranchConditional is checked in a similar spot. OpBranch seems to be unchecked currently.

Reply:
I could add support for OpBranch as well if wanted. That would likely be a separate issue / PR though.

Reply:
That's definitely an useful check if you don't mind adding it. Not a problem as a separate PR.